City-of-Helsinki/drupal-helfi-rekry
# UHF-X: Fix sonarcloud errors

# Fix sonarcloud errors
Fixes sonarcloud error and provides better error handling for initialization

## What was done
* Fixed an issue with sonar cloud (bad array destructuring)
* Developed better error handling to make app die with an error message in case initial data can't be retrieved

## How to install
* Get this branch: `git checkout origin/UHF-X-job-search-sonarcloud`
* Add this to your .env file: `ELASTIC_PROXY_URL=https://rekry-elastic-proxy-test.agw.arodevtest.hel.fi/job_listings` (use test data so you don't need to set up local elastic)
* Run `make fresh` or `make up drush-cr`

## How to test
* Add or edit a landing page to include a 'job search' paragraph
* Search should work, but if you...
  * Edit line `109` from `public/modules/custom/helfi_rekry_job_search/js/src/store.tsx` to be `const responses = {};`
  * Run `nvm use; npm i; npm run build` in the `public/modules/custom/helfi_rekry_job_search/js` directory
  * ... The app should die displaying an error message for the user
